content/docs/command-reference/stage/list.md

```diff
@@ -0,0 +1,80 @@
+# stage list
+
+List <abbr>stages</abbr> in the project.
+
+## Synopsis
+
+```usage
+usage: dvc stage list [-h] [-q | -v]
+                      [-R] [--all] [--fail] [--names-only]
+                      [targets ...]
+
+positional arguments:
+  targets          Show stages from a dvc.yaml file or a directory (with -R).
+                   'dvc.yaml' by default.
+```
+
+## Description
+
+`dvc stage list` is useful for quick listing of stages present in the project
+without having to open `dvc.yaml` files.
+
+It provides a list of stages, with a stage name and the description of the
+stage. The description of the stage is read from
+[`desc` field in the `dvc.yaml` file](doc/user-guide/project-structure/pipelines-files#stage-entries),
+or if it does not exist, DVC generates the short description from the other
+metadata it has about the stage.
+
+Without any targets, by default, it only lists the stages from the `dvc.yaml`
+present in the current working directory.
+
+DVC also uses this command to provide autocompletion in `bash` and `zsh`.
+
+## Options
+
+- `targets` (optional command argument) - stages to list from (`./dvc.yaml` by
+  default). Different things can be provided as targets depending on the flags
+  used (more details in each option). Examples:
+
+  - `dvc stage list linear/dvc.yaml`: Specific `dvc.yaml` file to list from
+  - `dvc stage list -R pipelines/`: Directory path to explore recursively for
+    `dvc.yaml` files
+  - `dvc stage list train-model`: Specific stage name or a
+    [foreach group](doc/user-guide/project-structure/pipelines-files#foreach-stages)
+    in `./dvc.yaml`
+  - `dvc stage list modeling/dvc.yaml:prepare`: Stage or a
+    [foreach group](doc/user-guide/project-structure/pipelines-files#foreach-stages)
+    in a specific `dvc.yaml` file
```

Comment:
Just one Q guys: I thought we didn't want the "group" term for `foreach` stages? BTW I'd be happy to adopt this but requires some changes to the dvc.yaml guide (for context).

Reply:
So in the example:

```yaml
stages:
  cleanups:
    foreach: # List of simple values
      - raw1
      - labels1
      - raw2
    do:
      cmd: clean.py "${item}"
      outs:
        - ${item}.cln
```

It looks like both of these are supported:
`dvc stage list cleanups`
`dvc stage list cleanups@raw1`

That's great!

Maybe we could describe `cleanups` as a `foreach key`, or is that too technical? Is this even worth documenting? It seems pretty intuitive to try `dvc stage list cleanups` in this scenario.

Reply:
"foreach key" seems a bit technical.

Maybe it's not worth being so explicit indeed, and we can just say "Stage(s)" (here and in https://dvc.org/doc/command-reference/repro#options)

> dvc stage list cleanups@raw1

That may need documenting though. We currently say "in dvc.yaml" and `cleanups@raw1` really exists in dvc.lock... ⌛

Reply:
See #2380.

Reply:
I mean #2369.

Reply:
Good point. It's confusing, but not sure whether it's worth it to worry about the specifics here. Is there even a use case for `dvc stage list cleanups@raw1`?

Reply:
🤷 maybe not but keep in mind this also applies to `repro`.